### PR TITLE
[smoke-test] Removing clean state dependency of some tests

### PR DESCRIPTION
**Pull request by wrwg**
### Description

Refines #2140 to fix flakiness.

### Test Plan

Existing tests, less flaky

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2143)

### CODE REVIEW

File: config/global-constants/src/lib.rs [hunk 1]
```diff
@@ -22,5 +22,5 @@ pub const VALIDATOR_NETWORK_KEY: &str = "validator_network";
 pub const SAFETY_DATA: &str = "safety_data";
 pub const WAYPOINT: &str = "waypoint";
 pub const GENESIS_WAYPOINT: &str = "genesis-waypoint";
-pub const MOVE_MODULES: &str = "move_modules";
+pub const MOVE_MODULES: &str = "module_publish_modules";
```
Comment (bchocho) [inline config/global-constants/src/lib.rs:25]: Is this an intended change? Seems out of scope of the smoke-test

File: testsuite/smoke-test/src/aptos/module_publish_modules/sources/hello.move [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        move_to(
+            sender,
+            ModuleData { global_counter: 0 }
+        );
+    }
+
+    public fun foo(addr: address): u64 {
+        coin::balance<AptosCoin>(addr)
+    }
+}
```
Comment (bchocho) [inline testsuite/smoke-test/src/aptos/module_publish_modules/sources/hello.move:19]: Some linter may complain about the newline.

Comment (areshand) [inline testsuite/smoke-test/src/aptos/module_publish_modules/sources/hello.move:19]: one more line

File: config/global-constants/src/lib.rs (hunk 1 above)
Comment (wrwg) [reply config/global-constants/src/lib.rs:25]: Oh! That was jetbrains refactor renaming...

**Review by bchocho — APPROVED** [review]
LGTM
